### ISSUE TITLE
build failure with rust nightly

### ISSUE THREAD

**Issue by ustime** [AlexPikalov/cdrs#142]
error[E0281]: type mismatch: `fn(std::io::Error) -> compression::CompressionError {compression::CompressionError::Lz4}` implements the trait `std::ops::FnOnce<(std::io::Error,)>`, but the trait `std::ops::FnOnce<(lz4_compress::decompress::Error,)>` is required
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/cdrs-1.0.0-beta.9/src/compression.rs:178:38
    |
178 |         lz4::decompress(&bytes[4..]).map_err(CompressionError::Lz4)
    |                                      ^^^^^^^ expected enum `lz4_compress::decompress::Error`, found struct `std::io::Error`

**Comment by TrionProg** [comment]
I come here to write about this error. I have on
```
rustc 1.17.0 (56124baa9 2017-04-24)
rustc 1.16.0 (30cf806ef 2017-03-10)
```
**Alles schlecht**

How to avoid this problem(urgently)?

**Comment by TrionProg** [comment]
Ok. I found easy and stupid solution - just edited source. Of course I should forget about lz4 =) But I hurry.
```
//lz4::decompress(&bytes[4..]).map_err(CompressionError::Lz4)
        Ok(Vec::new())
```

Note, that user must write in Cargo.toml
```
cdrs = "1.0.0-beta.9"
```
otherwise cargo will download 0.7.0 that do not knows what is `transport::TransportTCP`.

**Comment by BenjaminGill-Metaswitch** [comment]
I don't think it's a problem with nightly - I am also seeing it in stable. Rather, https://crates.io/crates/lz4-compress released a new version recently, and that seems to have broken things.

As a workaround, I think you can manually specify v0.1.0 of lz4-compress

**Comment by AlexPikalov** [comment]
@BenjaminGill-Metaswitch actually lz4-compress version is fixed to 0.1.0. Please let me check it

**Comment by AlexPikalov** [comment]
@TrionProg sorry for delay, I'll try to investigate this issue and come with something

**Comment by BenjaminGill-Metaswitch** [comment]
@AlexPikalov - I think `lz4-compress = "0.1.0"` allows any of `0.1.0 <= v < 0.2.0`. To fix it to 0.1.0, I think you probably need `lz4-compress = "~0.1.0"`.

See http://doc.crates.io/specifying-dependencies.html

**Comment by AlexPikalov** [comment]
@BenjaminGill-Metaswitch I've just checked cargo docs. Thats true that `0.1.0` allows all minor versions because it's supposed that minor version does not have breaking changes. Which is not the case for LZ4 `0.1.1`. The version with tilde won't work as well because of that. The working solution is `=0.1.0`. I've prepared related PR. Will let you know when new version of CDRS is released

**Comment by AlexPikalov** [comment]
Fixed in 1.0.0-rc.1
